Examples/Examples/2-Environment/EnvironmentDemoVC+Reducer.swift

```diff
@@ -92,4 +71,49 @@ let environmentReducer: Reducer<EnvironmentState, EnvironmentAction, Environment
             return .none
         }
     }
-}()
+}
+
+struct AnalyticsEvent: Equatable {
+    var name: String
+    var category: String
+}
+
+struct CustomError: Error, Equatable {
+    var message: String
+}
+
+class AnalyticsManager {
+    private init() {}
+    static func track(_ event: AnalyticsEvent) {
+        print(">> Tracked: \(event)")
+    }
+}
+
+struct EnvironmentVCEnvironment {
+    var loadData: () -> Effect<Result<Int, CustomError>>
+    var trackEvent: (AnalyticsEvent) -> Void
+    var date: () -> Date
+    var uuid: () -> UUID
+}
+
+private enum EnvironmentVCKey: DependencyKey {
+    static var liveValue: EnvironmentVCEnvironment {
+        EnvironmentVCEnvironment(
+            loadData: {
+                Observable.just(Result.success(Int.random(in: 0 ... 10000)))
+                    .delay(.milliseconds(500), scheduler: MainScheduler.instance)
+                    .eraseToEffect()
+            },
+            trackEvent: AnalyticsManager.track(_:),
+            date: Date.init,
+            uuid: UUID.init
+        )
+    }
+}
```

Comment:
this enum is not needed.
you can make it as 
```
extension EnvironmentVCEnvironment: DependencyKey {
     static var liveValue: Self {
         Self(
             loadData: {
                 Observable.just(Result.success(Int.random(in: 0 ... 10000)))
                     .delay(.milliseconds(500), scheduler: MainScheduler.instance)
                     .eraseToEffect()
             },
             trackEvent: AnalyticsManager.track(_:),
             date: Date.init,
             uuid: UUID.init
         )
     }
 }
```

Reply:
yep, updated here [2225b02](https://github.com/tokopedia/RxComposableArchitecture/pull/63/commits/2225b024ba72e3c04bf8526dc9be4747568bb1ef)